src/Core/src/Platform/iOS/MauiView.cs

```diff
@@ -143,16 +143,25 @@ SafeAreaRegions GetSafeAreaRegionForEdge(int edge)
 			return SafeAreaRegions.None;
 		}
 
-		static double GetSafeAreaForEdge(SafeAreaRegions safeAreaRegion, double originalSafeArea)
+		double GetSafeAreaForEdge(SafeAreaRegions safeAreaRegion, double originalSafeArea, int edge)
 		{
 			// Edge-to-edge content - no safe area padding
 			if (safeAreaRegion == SafeAreaRegions.None)
 				return 0;
 
+			// Handle SoftInput specifically - only apply padding when keyboard is actually showing
+			if (edge == 3 && SafeAreaEdges.IsSoftInput(safeAreaRegion))
```

Comment:
I don't think this check is quite right

It needs to validate if the user only wants SoftInput

For example, if I set the SafeAreaEdges to ALL it's no no longer working on the bottom edge because ALL means softinput and bottom edge. 

Since the tests on this one are currently green, please also add a test that would have caught this!